eip-6170.md

```diff
@@ -0,0 +1,107 @@
+    /// @param _receiver is the address of the receiver.
+    /// @param _toChainId is the identifier of the receiving blockchain.
+    /// @param _message is the arbitray message being sent.
+    /// NOTE: this emits when the message is sent, it does not gurantee successful delivery.
+    event MessageSent(
+        bytes _sender,
+        bytes _fromChainId,
+        bytes _receiver,
+        bytes _toChainId,
+        bytes _message
+    );
+
+    /// @dev This emits when a cross-chain message is delivered succesfully at the receiver.
+    /// @param _sender is the address of the sender.
+    /// @param _fromChainId is the identifier of the sending blockchain.
+    /// @param _receiver is the address of the receiver.
+    /// @param _toChainId is the identifier of the receiving blockchain.
+    /// @param _message is the arbitray message being sent.
+    /// NOTE: this emits when the message is delivered successfully.
+    event MessageReceived(
+        bytes _sender,
+        bytes _fromChainId,
+        bytes _receiver,
+        bytes _toChainId,
+        bytes _message
+    );
+
+    /// @dev Sends message to a receiving address on different blockchain.
+    /// @param chainId is the unique identifier of receiving blockchain.
+    /// @param receiver is the address of the receiver.
+    /// @param message is the arbitrary message to be delivered.
+    /// @return the status of the process on the sending chain.
+    /// Note: this function is designed to support both evm and non-evm chains
+    /// Note: proposing chain-ids be the bytes encoding of their native token name string. For eg., abi.encode("ETH"), abi.encode("SOL") imagining they cannot override.
+    /// Note: MUST emit a {MessageSent} event.
+    function sendMessage(
+        bytes memory chainId,
+        bytes memory receiver,
+        bytes memory message
+    ) external returns (bool);
+
+    /// @dev Receives message from a sender on different blockchain.
+    /// @param chainId is the unique identifier of the sending blockchain.
+    /// @param sender is the address of the sender.
+    /// @param message is the arbitrary message sent by the sender.
+    /// @return the status of message processing/storage.
+    /// Note: MUST emit a {MessageReceived} event.
+    /// Note: sender validation (or) message validation should happen before processing the message.
+    function receiveMessage(
+        bytes memory chainId,
+        bytes memory sender,
+        bytes memory message
+    ) external returns (bool);
+}
+```
+
+## Rationale
+
+The Cross-Chain interface is designed to be optimized for messaging layer integrators with a feature complete yet minimal interface. Validations such as sender authentication, receiver whitelisting, relayer mechanisms and cross-chain execution overrides are intentionally not specified, as Messaging protocols are expected to be treated as black boxes on-chain and inspected off-chain before use.
```

Comment:
"interoperability layer" from "messaging layer" (the more frequently used term to describe these)
"feature-complete, yet" from "feature complete yet"

```suggestion
The Cross-Chain interface is designed to be optimized for interoperability layer integrators with a feature-complete, yet minimal interface. Validations such as sender authentication, receiver whitelisting, relayer mechanisms and cross-chain execution overrides are intentionally not specified, as Messaging protocols are expected to be treated as black boxes on-chain and inspected off-chain before use.
```